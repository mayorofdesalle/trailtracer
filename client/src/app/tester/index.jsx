import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
