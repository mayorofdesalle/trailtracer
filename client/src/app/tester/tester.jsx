import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };