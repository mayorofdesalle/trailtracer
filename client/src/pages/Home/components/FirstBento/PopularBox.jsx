import styled from 'styled-components';

import BentoBox from '@components/ui/BentoBox';
import { glassMorph } from '@components/misc/mixins';
import { rotateInFB } from '@components/misc/anims';

import { pseudoBentoArrowBottom } from '../mixins';

/**
 * PopularBox
 * @description
 * This is a styled bento box that is used to display the most popular trails.
 **/
const PopularBox = styled(BentoBox)`
    outline: 2px solid ${({ theme }) => theme.colors.primaryTransparent};
    animation: ${rotateInFB} 1s ease-in-out;
    
    --arrowColor: ${({ theme }) => theme.colors.secondary};
    ${pseudoBentoArrowBottom}
    ${glassMorph}
`;

export default PopularBox;