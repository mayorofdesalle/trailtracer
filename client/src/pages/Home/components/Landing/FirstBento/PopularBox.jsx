import styled from 'styled-components';

import BentoBox from '@components/ui/Bento/Box';
import { pseudoBentoArrowBottom } from '@components/misc/mixins';
import { slideFromTop } from '@components/misc/anims';

/**
 * PopularBox
 * @description
 * This is a styled bento box that is used to display the most popular trails.
 **/
const PopularBox = styled(BentoBox)`
    grid-area: PopularBox;
    --arrowColor: ${({ theme }) => theme.colors.secondary};
    ${pseudoBentoArrowBottom}

    &::before {
        animation: ${slideFromTop} 1s ease-in-out;
    }
`;

export default PopularBox;