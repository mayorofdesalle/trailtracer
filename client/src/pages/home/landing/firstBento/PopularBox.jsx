import styled from 'styled-components';

import { slideFromTop } from '@components/anims';
import { BentoBox } from '@components/ui/bento';
import { pseudoBentoArrowBottom } from '@components/ui/bento/mixins';

/**
 * PopularBox
 * @description
 * Styled bento box to display the most popular trails.
 **/
const PopularBox = styled(BentoBox)`
    grid-area: PopularBox;
    --arrowColor: ${({ theme }) => theme.colors.secondary};
    ${pseudoBentoArrowBottom}

    &::before {
        animation: ${slideFromTop} 1s ease-in-out;
    }
`;

export default PopularBox;