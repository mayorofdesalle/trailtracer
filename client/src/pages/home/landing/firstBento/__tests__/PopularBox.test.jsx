import { describe, expect, it, render, screen } from '@app/tester';

import PopularBox from '../PopularBox';

describe('PopularBox', () => {

    it('should render correctly', () => {
        render(<PopularBox data-testid='popular-box' />);
        const popularBox = screen.getByTestId('popular-box');

        expect(popularBox).toBeInTheDocument();
        expect(popularBox).toHaveStyle('grid-area: PopularBox');
    });

});