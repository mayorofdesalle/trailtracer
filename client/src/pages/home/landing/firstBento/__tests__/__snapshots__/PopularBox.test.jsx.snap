// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopularBox > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beySPh sc-guDLey sc-hLQSwg jVdZA lbUchL kjbnwM"
  />
</DocumentFragment>
`;
